File: Semester 2/Research Project/2020_AON01/Stage_API/Stage_API.Domain/Classes/Stagevoorstel.cs
using Stage_API.Domain.enums;
using Stage_API.Domain.Relations;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace Stage_API.Domain.Classes
{
    public class Stagevoorstel
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public Bedrijf Bedrijf { get; set; }
        [IgnoreDataMember]
        public int? BedrijfId { get; set; }
        /*HIER IS DE MANY-MANY*/
        public ICollection<ReviewerStagevoorstelFavoriet> ReviewersFavorieten { get; set; }
        public ICollection<ReviewerStagevoorstelToegewezen> ReviewersToegewezen { get; set; }
        public ICollection<StudentStagevoorstelFavoriet> StudentenFavorieten { get; set; }
        public ICollection<Student> StudentenToegewezen { get; set; }
        public string Titel { get; set; }
        public string Adres { get; set; }
        public string StagePostcode { get; set; }
        public string Gemeente { get; set; }
        public int StageITMedewerkers { get; set; }
        public string[] AfstudeerrichtingVoorkeur { get; set; }
        public string OpdrachtOmschrijving { get; set; }
        public string[] Omgeving { get; set; }
        public string OmgevingOmschrijving { get; set; }
        public string Randvoorwaarden { get; set; }
        public string Onderzoeksthema { get; set; }
        public string[] Verwachtingen { get; set; }
        public int AantalGewensteStagiairs { get; set; }
        //Bedrijf is gecontacteerd door student(en) en wenst deze opdracht enkel aan deze student(en) aan te bieden.
        public string GereserveerdeStudenten { get; set; }
        public string Bemerkingen { get; set; }
        //Semester 1 (oktober - januari) OF Semester 2 (februari - juni)
        public int Periode { get; set; }
        public BeoordelingStatus Status { get; set; }
        public ICollection<Review> Reviews { get; set; }
        public ICollection<Comment> Comments { get; set; }
    }
}

File: Semester 2/Research Project/2020_AON01/Stage-Frontend/src/app/classes/stagevoorstel.ts
import { BeoordelingStatus } from './enums/beoordeling-status.enum';
import { Review } from './review';
import { Bedrijf } from './bedrijf';
import { ReviewerStagevoorstelFavoriet } from './reviewer-stagevoorstel-favoriet';
import { ReviewerStagevoorstelToegewezen } from './reviewer-stagevoorstel-toegewezen';
import { Contactpersoon } from './contactpersoon';
import { Bedrijfspromotor } from './bedrijfspromotor';
import { StudentStagevoorstelFavoriet } from './student-stagevoorstel-favoriet';
import { StudentStagevoorstelToegewezen } from './student-stagevoorstel-toegewezen';
import { Reviewer } from '@app/classes/reviewer';
import { Student } from '@app/classes/student';
import { Comment } from '@app/classes/comment';

export class Stagevoorstel {
  id: number;
  date: string;
  bedrijf: Bedrijf;
  reviewersFavoriet: Reviewer[];
  reviewersToegewezen: Reviewer[];
  studentenFavorieten: Student[];
  studentToegewezen: Student[];
  titel: string;
  stageStraat: string;
  stageNummer: string;
  stagePostcode: string;
  gemeente: string;
  stageITMedewerkers: number;
  contactpersoon: Contactpersoon;
  bedrijfspromotor: Bedrijfspromotor;
  afstudeerrichtingVoorkeur: string[];
  opdrachtOmschrijving: string;
  omgeving: string[];
  omgevingOmschrijving: string;
  randvoorwaarden: string;
  onderzoeksthema: string;
  verwachtingen: string[];
  aantalGewensteStagiairs: number;
  gereserveerdeStudenten: string;
  bemerkingen: string;
  periode: number;
  status: BeoordelingStatus;
  reviews: Review[];
  comments: Comment[];

  // stagevoorstelModelStudent
  bedrijfsNaam: string;
  studentenFavorietenCount: number;
}
